cms/djangoapps/contentstore/rest_api/v0/tests/test_advanced_settings.py

```diff
@@ -0,0 +1,64 @@
+""" Tests for the course advanced settings API. """
+import json
+
+import ddt
+from django.test import override_settings
+from django.urls import reverse
+from milestones.tests.utils import MilestonesTestCaseMixin
+
+from cms.djangoapps.contentstore.tests.utils import CourseTestCase
+
+
+@ddt.ddt
+class CourseAdvanceSettingViewTest(CourseTestCase, MilestonesTestCaseMixin):
+    """
+    Tests for AdvanceSettings API View.
+    """
+
+    def setUp(self):
+        super().setUp()
+        self.url = reverse(
+            "cms.djangoapps.contentstore:v0:course_advanced_settings",
+            kwargs={"course_id": self.course.id},
+        )
+
+    def get_and_check_developer_response(self, response):
+        """
+        Make basic asserting about the presence of an error response, and return the developer response.
+        """
+        content = json.loads(response.content.decode("utf-8"))
+        assert "developer_message" in content
+        return content["developer_message"]
+
+    def test_permissions_unauthenticated(self):
+        """Test that an error is returned in the absence of auth credentials."""
+        self.client.logout()
+        response = self.client.get(self.url)
+        error = self.get_and_check_developer_response(response)
+        assert error == "Authentication credentials were not provided."
+
+    def test_permissions_unauthorized(self):
+        """Test that an error is returned if the user is unauthorised."""
+        client, _ = self.create_non_staff_authed_user_client()
+        response = client.get(self.url)
+        error = self.get_and_check_developer_response(response)
+        assert error == "You do not have permission to perform this action."
+
+    @ddt.data(
+        ("ENABLE_EDXNOTES", "edxnotes"),
+        ("ENABLE_OTHER_COURSE_SETTINGS", "other_course_settings"),
+    )
+    @ddt.unpack
+    def test_conditionally_excluded_fields_not_present(self, setting, excluded_field):
+        """
+        Test that the response doesn't contain fields excluded on the bases of enabled/disabled features.
+        """
+        with override_settings(FEATURES={setting: True}):
```

Comment:
how this is a valid dict syntax? `{setting: True}` shouldn't this be `{'setting': True}`? Am I missing something?

Also, can you point me to env setting in common.py which this is overrding? 
thanks.

Reply:
By putting quotes around 'setting' I would be making it into a string with the value of "setting", that is not the intention here. 

Here is an example, where we want to create a dictionary of usernames to grades:

```python
username = 'kshitij'
grade = 57.0

user_grades = {
    username: grade,
}
```
This will produce a dictionary with the value:

```python
user_grades = {
    'kshitij': 57.0,
}
```
i.e. if you use variable as the key name its value will be substituted in place. 

In this example, we are using `ddt` to pass different values of `setting` and `excluded_field` by adding the decorator:

```python
 @ddt.data(
    ("ENABLE_EDXNOTES", "edxnotes"),
    ("ENABLE_OTHER_COURSE_SETTINGS", "other_course_settings"),
)
@ddt.unpack
```
So first it will run the test with `setting = "ENABLE_EDXNOTES"` and `excluded_field = "ENABLE_OTHER_COURSE_SETTINGS"`. 

So, the test with override as:
```python
        with override_settings(FEATURES={"ENABLE_EDXNOTES": True}):
```

Reply:
Oh right, sorry I was not paying keen intention on the `setting` is a ddt variable. sorry for the silly question. we are good here.